vm-setup/vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml

```diff
@@ -0,0 +1,153 @@
+        line: "deb [signed-by=/usr/share/keyrings/libcontainers-archive-keyring.gpg] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_{{ OS_VERSION_ID }}/ /"
+        create: yes
+        owner: root
+        group: root
+        mode: '0644'
+      when: ansible_distribution_release == "noble"
+
+    - name: Update all packages to their latest version
+      apt:
+        name: "*"
+        state: latest
+
+    - name: Podman
+      block:
+        - name: Install podman
+          apt: 
+            name: podman
+            state: present
+
+        - name: Registries configuration for podman
+          blockinfile:
+            path: /etc/containers/registries.conf
+            block: |
+                [registries.insecure]
+                registries = ['{{ REGISTRY }}']
+      become: yes
+      when: CONTAINER_RUNTIME == "podman"
+
+    - name: Install docker
+      block:
+        - name: Create /etc/apt/keyrings folder
+          file:
+            path: /etc/apt/keyrings
+            state: directory
+
+        - name: Add Docker’s GPG key
+          get_url:
+            url: https://download.docker.com/linux/ubuntu/gpg
+            dest: /etc/apt/keyrings/docker.asc
+            mode: '0644'
+            force: true
+          
+        - name: Dearmor GPG key
+          shell: | 
+            cat /etc/apt/keyrings/docker.asc | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+                
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository (without gpg key file location) 
+          apt_repository:
+            repo: "deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+            state: absent
+
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository (without gpg key file location) 
+          apt_repository:
+            repo: "deb  https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+            state: absent
+
+        - name: Add Docker Repository
+          apt_repository:
+            repo: "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+            state: present
+
+        - name: Update all packages to their latest version
+          apt:
+            name: "*"
+            state: latest
+
+        - name: Install docker
+          apt: name={{ item }} state=latest update_cache=yes
+          loop: [ 'docker-ce', 'docker-ce-cli', 'containerd.io' ]
+
+        - name: Create docker configuration dir
+          file:
+            path: /etc/docker
+            state: directory
+            owner: root
+            group: root
+
+        - name: Template daemon.json to /etc/docker/daemon.json
+          template:
+            src: "{{ DAEMON_JSON_PATH }}/daemon.json"
```

Comment:
This one looks weird.
Instead of a template, this lives under files, but we pass the path to it as an environment variable.